### PR TITLE
Add model receive shadows option to 3d puck config

### PR DESCRIPTION
**Pull request by evil159**
### Summary of changes

This PR adds an experimental option to control shadow receiving for the 3D model puck - `LocationPuck3D.modelReceiveShadows `.

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [x] Document any changes to public APIs.
 - [] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [x] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: app/src/main/java/com/mapbox/maps/testapp/examples/LocationComponentActivity.kt [hunk 1]
```diff
@@ -200,10 +200,11 @@ class LocationComponentActivity : AppCompatActivity() {
         )
         is LocationPuck2D -> it.locationPuck = LocationPuck3D(
           modelUri = "asset://sportcar.glb",
-          modelScale = listOf(0.1f, 0.1f, 0.1f),
+          modelScale = listOf(0.5f, 0.5f, 0.5f),
```
[the remark this answers is not in the document]
Comment (evil159) [reply app/src/main/java/com/mapbox/maps/testapp/examples/LocationComponentActivity.kt]: Supposed to be debug-only change 🙈 Reverted

**Review by kiryldz — APPROVED** [review]
LGTM but would appreciate 👀 from @pengdev who was adding similar API so that nothing is missed. Also I've added checkbox that this PR needs to be ported to v10.7 branch but someone from @mapbox/maps-android could handle that 😄

**Review by ank27 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,13 @@ Mapbox welcomes participation and contributions from everyone.
 * Fix excessive logs appearing sometimes after `onStop` lifecycle event. ([1527](https://github.com/mapbox/mapbox-maps-android/pull/1527))
 * Fix `com.mapbox.maps.MapboxMapException` crash on style load. ([1532](https://github.com/mapbox/mapbox-maps-android/pull/1532))
 
+# main
```
Comment (yunikkk) [inline CHANGELOG.md:8]:
```suggestion
# main
```

Comment (yunikkk) [inline CHANGELOG.md:8]: This one is not needed

File: CHANGELOG.md [hunk 3]
```diff
@@ -64,6 +64,13 @@ Mapbox welcomes participation and contributions from everyone.
   Also bumps [WorkManager 2.7.1](https://developer.android.com/jetpack/androidx/releases/work#2.7.1) that enforces compileSdk 31 or newer.
 * Bump gl-native to v10.7.0, common to 22.1.0. ([1543](https://github.com/mapbox/mapbox-maps-android/pull/1543))
 
+# main
```
Comment (yunikkk) [inline CHANGELOG.md:67]: Need to merge with 10.7.0 section and remove `#main`

**Pull request closed without merging**